libvast/src/logger.cpp

```diff
@@ -68,25 +58,191 @@ int loglevel_to_int(caf::atom_value x, int default_value) {
       return VAST_LOG_LEVEL_DEBUG;
     case caf::atom_uint("trace"):
       return VAST_LOG_LEVEL_TRACE;
+    default:
+      return default_value;
+  }
+}
+
+namespace {
+
+constexpr bool is_vast_loglevel(const int value) {
+  switch (value) {
+    case VAST_LOG_LEVEL_QUIET:
+    case VAST_LOG_LEVEL_ERROR:
+    case VAST_LOG_LEVEL_WARNING:
+    case VAST_LOG_LEVEL_INFO:
+    case VAST_LOG_LEVEL_VERBOSE:
+    case VAST_LOG_LEVEL_DEBUG:
+    case VAST_LOG_LEVEL_TRACE:
+      return true;
   }
+  return false;
+}
+
+/// Converts a vast log level to spdlog level
+spdlog::level::level_enum vast_loglevel_to_spd(const int value) {
+  VAST_ASSERT(is_vast_loglevel(value));
+  spdlog::level::level_enum level = spdlog::level::off;
+  switch (value) {
+    case VAST_LOG_LEVEL_QUIET:
+      break;
+    case VAST_LOG_LEVEL_ERROR:
+      level = spdlog::level::err;
+      break;
+    case VAST_LOG_LEVEL_WARNING:
+      level = spdlog::level::warn;
+      break;
+    case VAST_LOG_LEVEL_INFO:
+      level = spdlog::level::info;
+      break;
+    case VAST_LOG_LEVEL_VERBOSE:
+      level = spdlog::level::debug;
+      break;
+    case VAST_LOG_LEVEL_DEBUG:
+      level = spdlog::level::debug;
```

Comment:
DIdn't we say that this was supposed to be trace? (might be wrong here, but please double-check)